swc/private/swc.bzl

```diff
@@ -115,88 +115,109 @@ def _calculate_map_outs(srcs, source_maps, out_dir = None, root_dir = None):
     return [f2 for f2 in [_calculate_map_out(f, source_maps, out_dir, root_dir) for f in srcs] if f2]
 
 def _impl(ctx):
-    swcinfo = ctx.toolchains["@aspect_rules_swc//swc:toolchain_type"].swcinfo
-    env = {
-        # Our patch for @swc/core uses this environment variable to locate the rust binding
-        "SWC_BINARY_PATH": "../../../" + swcinfo.binding,
-        "BAZEL_BINDIR": ctx.bin_dir.path,
-    }
+    swc_toolchain = ctx.toolchains["@aspect_rules_swc//swc:toolchain_type"]
+
+    inputs = swc_toolchain.swcinfo.tool_files[:]
 
     args = ctx.actions.args()
+    args.add("compile")
 
     # Add user specified arguments *before* rule supplied arguments
     args.add_all(ctx.attr.args)
-    args.add("--source-maps", ctx.attr.source_maps)
+
+    args.add_all(["--source-maps", ctx.attr.source_maps])
 
     if ctx.attr.output_dir:
         if len(ctx.attr.srcs) != 1:
             fail("Under output_dir, there must be a single entry in srcs")
         if not ctx.files.srcs[0].is_directory:
             fail("Under output_dir, the srcs must be directories, not files")
-        output_dir = ctx.actions.declare_directory(ctx.attr.out_dir if ctx.attr.out_dir else ctx.label.name)
 
-        output_sources = [output_dir]
+        inputs.extend(ctx.files.srcs)
 
+        output_dir = ctx.actions.declare_directory(ctx.attr.out_dir if ctx.attr.out_dir else ctx.label.name)
         args.add_all([
-            ctx.files.srcs[0].short_path,
             "--out-dir",
-            output_dir.short_path,
-            "--no-swcrc",
-            "-q",
+            output_dir.path,
         ])
+        output_sources = [output_dir]
 
-        ctx.actions.run(
-            inputs = copy_files_to_bin_actions(ctx, ctx.files.srcs) + swcinfo.tool_files,
-            arguments = [args],
-            outputs = [output_dir],
-            env = env,
-            executable = ctx.executable.swc_cli,
-            progress_message = "Transpiling with swc %{label}",
+        # FIXME: unused?
```

Comment:
@realtimetodie I think we're missing a test here for the directory output case that an swcrc can be used?